src/languages/javascript.js

```diff
@@ -113,7 +113,19 @@ function(hljs) {
           contains : [
             {
               className : 'doctag',
-              begin : '@[A-Za-z]+(\\s+\\{[A-Za-z0-9><\\._:-]+\\}(\\s+[A-Za-z0-9\\._:-]+(?=\\s+-))?)?'
+              begin : '@[A-Za-z]+\\s+',
+              contains : [
+                {
+                  className: 'type',
+                  relevance: 0,
+                  begin: '\\{',
+                  end: '\\}'
+                }
+                , {
+                  className: 'variable',
+                  begin: '\\s+' + IDENT_RE + '(?=\\s*(-)|($))',
```

Comment:
Am I correct this would fail with:

```
 * @param {boolean} retArr If set to true, the function will return an array
```

A sample from:
https://devdocs.io/jsdoc/tags-returns

Reply:
Yes, but do you have idea how to identify variable name? We don't want to have it for 
@return {Point} A Point object

I used - or end of line as information for variable name. The other solution is to parse @param and only there make that it can have variable.

Reply:
Yeah, I'm torn.  I can just see someone filing an issue about this exact issue, but then if you fix @param you COULD end up down the rabbit hole of people expecting us to support ALL the tags:

https://devdocs.io/jsdoc-tags/

Which all seem to have their own set of rules...

Reply:
Won't the variable rule misfire though anytime you have a:

```
@something word
```

Pattern?  Wouldn't word match as a variable? (or some variant of that)

Reply:
I think as a starting point this is good enough. Maybe next feature/improvement will be to create a new jsdoc language which will be used in typescript, javascript. What do you think?

Reply:
I could be persuaded.  I think I would change the variable rule to use word boundaries for begin though and add a smaller rule to eat the spaces... that way the extra “ “ isn’t inside the variable span.  Usually we try to be clean about little stuff like that.  

I think after that we could call this done for starters.

I’m not sure about a whole “jsdoc” language, but I take your point about abstracting it into a component we can reuse in the various languages.  What is the full list?  JS, TypeScript, Java? More?

Reply:
Ok, it's now skipping spaces. 

Java is using JavaDoc which is a little bit different, for example you don't need to write types in JavaDoc. So JsDoc is important mostly for javascript, typescript, maybe in html. 

Let me know if this change is in good shape to be merged.